substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/pecoff/cv/CVTypeRecord.java

```diff
@@ -235,4 +413,806 @@ public boolean equals(Object obj) {
+            int pos = initialPos;
+            for (MDef f : methods) {
+                pos = f.computeContents(buffer, pos);
+            }
+            return pos;
+        }
+
+        @Override
+        public int hashCode() {
+            int h = type;
+            h = 31 * h + methods.hashCode();
+            return h;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (!super.equals(obj)) {
+                return false;
+            }
+            CVTypeMethodListRecord other = (CVTypeMethodListRecord) obj;
+            return this.methods.equals(other.methods);
+        }
+
+        @Override
+        public String toString() {
+            return String.format("LF_METHODLIST idx=0x%04x count=%d", getSequenceNumber(), methods.size());
+        }
+    }
+
+    static String attrString(short attrs) {
+        final StringBuilder sb = new StringBuilder();
+
+        /* Low byte. */
+        if ((attrs & MPROP_PPP_MASK) != 0) {
+            String[] aStr = {"", "private", "protected", "public"};
+            sb.append(aStr[attrs & MPROP_PPP_MASK]);
+        }
+        if ((attrs & MPROP_VSF_MASK) != 0) {
+            int p = (attrs & MPROP_VSF_MASK) >> 2;
+            String[] pStr = {"", " virtual", " static", " friend", " intro", " pure", " intro-pure", " (*7*)"};
+            sb.append(pStr[p]);
+        }
+        if ((attrs & MPROP_PSEUDO) != 0) {
+            sb.append(" pseudo");
+        }
+        if ((attrs & MPROP_FINAL_CLASS) != 0) {
+            sb.append(" final-class");
+        }
+        if ((attrs & MPROP_ABSTRACT) != 0) {
+            sb.append(" abstract");
+        }
+        if ((attrs & MPROP_COMPGENX) != 0) {
+            sb.append(" compgenx");
+        }
+        if ((attrs & MPROP_FINAL_METHOD) != 0) {
+            sb.append(" final-method");
+        }
+        return sb.toString();
+    }
+
+    abstract static class FieldRecord {
+
+        protected final short type;
+        protected final short attrs; /* property attribute field (prop_t) */
+        protected final String name;
+
+        protected FieldRecord(short leafType, short attrs, String name) {
+            assert name != null;
+            this.type = leafType;
+            this.attrs = attrs;
+            this.name = name;
+        }
+
+        protected FieldRecord(short leafType) {
+            this.type = leafType;
+            this.attrs = 0;
+            this.name = "";
+        }
+
+        public int computeSize() {
+            return computeContents(null, 0);
+        }
+
+        public abstract int computeContents(byte[] buffer, int initialPos);
+
+        @Override
+        public int hashCode() {
+            int h = type;
+            h = 31 * h + attrs;
+            h = 31 * h + name.hashCode();
+            return h;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (!super.equals(obj)) {
```

Comment:
While investigating the issue with the `LF_FIELDLIST` records, I noticed that this actually calls `Object.equals`, which is probably not what you intended.

Reply:
Nope, thanks for the catch.